BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -600,25 +600,37 @@ TpmIndicateReadyToBoot (
   in PCR 0 with EV_POST_CODE event type.
 
   @param[in] ComponentType    Stage whose measurement need to be extended.
+  @param[in] Signature        Signature of the component
+  @param[in] HashExtend       Component hash to be extended
 
   @retval RETURN_SUCCESS      Operation completed successfully.
   @retval Others              Unable to extend stage hash.
 **/
 RETURN_STATUS
 TpmExtendStageHash (
-  IN       UINT8            ComponentType
+  IN       UINT8            ComponentType,
+  IN       UINT32           Signature,
```

Comment:
how Signature get used here ?

Reply:
Signature is added for future implementations.
Ex. Signing is done with SHA256 hash and hashstore would be updated for same.
In measured boot, if we need to support SM3 alg, this API TpmExtendStageHash  would calculate SM3 hash and extend. Signature is used with GetComponetInfo() 

GetComponentInfo (**Signature**, &Src, &Length);

Reply:
If you have ComponentType, why need Component signature ?

Reply:
@mauricema I understand GetComponentInfo would use the Signature.
Does it supports with component type?

Reply:
Also, what is the purpose for 'HashExtend' argument?
If we need a API for directly extending a hash, we should probably add a new API rather then re-using this API.

Reply:
This is intended to extend hash provided instead of  retrieving from hashstore. 
As recommended, we could have an separate API to extend Hash directly.